### PR TITLE
feat#2927: now orphanet has auto ingestion and inheritence fields

### PR DESCRIPTION
**Pull request by botekchristophe**

### CODE REVIEW

File: src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  override val database = "variant"
+  override val tableName = "orphanet_gene_set"
+
+  override def extract(input: String)(implicit spark: SparkSession): Map[String, DataFrame] = {
+    import spark.implicits._
+
+    def loadXML: String => Elem = str => XML.loadString(spark.read.text(str).collect().map(_.getString(0)).mkString("\n"))
+
+    Map(
+      "product6" -> parseProduct6XML(loadXML(input + "/en_product6.xml")).toDF,
```
Comment (evans-g-crsj) [inline src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala:20]: Personally, I would favour string interpolation

File: src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  override def load(data: DataFrame, output: String)(implicit spark: SparkSession): DataFrame = {
+    data
+      .coalesce(1)
+      .write
+      .mode("overwrite")
+      .format("parquet")
+      .option("path", s"$output/$tableName")
+      .saveAsTable(s"$database.$tableName")
+
+    if (runEnv.equals(Environment.PROD))
```
Comment (evans-g-crsj) [inline src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala:42]: Personally, I would go with == vs equals

File: src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        (disorder \\ "AverageAgeOfOnsetList" \ "AverageAgeOfOnset" \ "Name").map(_.text).toList,
+        (disorder \\ "AverageAgeOfDeathList" \ "AverageAgeOfDeath" \ "Name").map(_.text).toList,
+        (disorder \\ "TypeOfInheritanceList" \ "TypeOfInheritance" \ "Name").map(_.text).toList
+      )
+    }
+  }
+
+}
+
+case class Product6(disorder_id: Long,
```
Comment (evans-g-crsj) [inline src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala:139]: I don't know the business logic needed, but the naming is not very telling in my opinion. "Product6" what does it represent?

File: src/main/scala/org/kidsfirstdrc/dwh/updates/UpdateVariant.scala [hunk 4]
```diff
@@ -48,10 +48,4 @@ class UpdateVariant(runEnv: Environment) extends MultiSourceEtlJob(runEnv) {
     if (runEnv.equals(Environment.PROD)) publishTable(releaseId_datetime, Variants.TABLE_NAME)
```
Comment (evans-g-crsj) [inline src/main/scala/org/kidsfirstdrc/dwh/updates/UpdateVariant.scala:48]: same as comment above

File: src/test/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetProductSpec.scala [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    product6DF.count shouldBe 3
+    product6DF.where($"orpha_code" === 447).as[OrphanetProduct6].collect().head shouldBe expectedProduct6
+
+    val expectedProduct9 = OrphanetProduct9()
+    product9DF.show(false)
+    product9DF.count shouldBe 2
+    product9DF.where($"orpha_code" === 58).as[OrphanetProduct9].collect().head shouldBe expectedProduct9
+  }
+
+  "transform" should "return an joined data from product 6 and 9" in {
```
Comment (evans-g-crsj) [inline src/test/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetProductSpec.scala:33]: a vs an

File: src/test/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetProductSpec.scala [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    import spark.implicits._
+
+    val path = this.getClass.getResource("/raw/orphanet").getFile
+    //val path = this.getClass.getResource("/full").getFile
```
Comment (evans-g-crsj) [inline src/test/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetProductSpec.scala:38]: is that comment useful?

File: src/test/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetProductSpec.scala [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    val outputDF = job.transform(extractedData)
+
+    outputDF.show(false)
+    outputDF.where($"orpha_code" === 166024).as[OrphanetOutput].collect().head shouldBe OrphanetOutput()
+
+    outputDF.as[OrphanetOutput].count() shouldBe 3
+
+  }
+
+  "load" should "create an hive table" in {
```
Comment (evans-g-crsj) [inline src/test/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetProductSpec.scala:52]: a vs an

File: src/test/scala/org/kidsfirstdrc/dwh/testutils/ClassGeneratorMain.scala [hunk 8]
```diff
@@ -15,13 +17,28 @@ object ClassGeneratorMain extends App with WithSparkSession {
     .withColumn("sampleId", lit("id"))
     .withColumn("genotypes", array(struct(col("sampleId") as "sampleId")))
     .drop("sampleId")
-  val clinvarOutput = ImportClinVar.transform(clinvarInput)
 
+  clinvarInput
+    //.writeCLassFile("org.kidsfirstdrc.dwh.testutils.external","ClinvarInput", root)
```
Comment (evans-g-crsj) [inline src/test/scala/org/kidsfirstdrc/dwh/testutils/ClassGeneratorMain.scala:22]: are these comments helpful?

File: src/test/scala/org/kidsfirstdrc/dwh/testutils/ClassGeneratorMain.scala [hunk 9]
```diff
@@ -15,13 +17,28 @@ object ClassGeneratorMain extends App with WithSparkSession {
...
-  val variants = spark.read.format("parquet").load("src/test/resources/variants/variants.parquet")
 
-  ClassGenerator.writeCLassFile("org.kidsfirstdrc.dwh.testutils.variant","Variant", variants, root)
+  spark.read.format("parquet").load("src/test/resources/variants/variants.parquet")
+    //.writeCLassFile("org.kidsfirstdrc.dwh.testutils.variant","Variant", root)
+
+
+  val orphanetPath = getClass.getResource("/raw/orphanet").getFile
+  val orphanetData = new ImportOrphanetJob(Environment.DEV).extract(orphanetPath)(spark)
+  orphanetData("product6").where(col("orpha_code") === 447)
```
Comment (evans-g-crsj) [inline src/test/scala/org/kidsfirstdrc/dwh/testutils/ClassGeneratorMain.scala:35]: magic numbers

**Review by evans-g-crsj — APPROVED** [review]

File: src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala (hunk 3 above)
Comment (jecos) [reply src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala:139]: This is the name of the files. I think it's my bad, I sent the link without giving any indications about these files. Product6 === Genes associated with rare diseases
Product9 === Rare diseases natural history

**Review by jecos — COMMENTED** [review]
Agree with comments from Evans

File: src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala (hunk 3 above)
Comment (botekchristophe) [reply src/main/scala/org/kidsfirstdrc/dwh/external/ImportOrphanetJob.scala:139]: I agree, I'll change these

File: src/test/scala/org/kidsfirstdrc/dwh/testutils/ClassGeneratorMain.scala (hunk 9 above)
Comment (botekchristophe) [reply src/test/scala/org/kidsfirstdrc/dwh/testutils/ClassGeneratorMain.scala:35]: I want to find exactly this id in my test data or the test should fail

**Pull request merged**